packages/babel-helpers/src/helpers/applyDecs2305.ts

```diff
@@ -674,32 +189,407 @@ function defineMetadata(Class: any, metadata: any) {
+              throw new TypeError(
+                "accessor decorators must return an object with get, set, or init properties or void 0",
+              );
+            }
+          } else if (
+            assertCallable(
+              newValue,
+              (isField ? "field" : "method") + " decorators",
+              "return",
+            )
+          ) {
+            if (isField) {
+              init.push(newValue);
+            } else {
+              desc[key] = newValue;
+            }
+          }
+        }
+      } finally {
+        decoratorFinishedRef.v = true;
+      }
+    }
+
+    if (isField || isAccessor) {
+      ret.push(function (instance: any, value: any) {
+        for (var i = init.length - 1; i >= 0; i--) {
+          value = init[i].call(instance, value);
+        }
+        return value;
+      });
+    }
+
+    if (!isField && !isClass) {
+      if (isPrivate) {
+        if (isAccessor) {
+          ret.push(_bindPropCall(desc, "get"), _bindPropCall(desc, "set"));
+        } else {
+          ret.push(
+            kind === PROP_KIND.METHOD
+              ? desc[key]
+              : _bindPropCall.call.bind(desc[key]),
+          );
+        }
+      } else {
+        Object.defineProperty(Class, name, desc);
+      }
+    }
+    return newValue;
+  }
+
+  /* @no-mangle */
+  function applyMemberDecs(
+    Class: any,
+    decInfos: DecoratorInfo[],
+    instanceBrand: Function,
+    metadata: any,
+  ) {
+    var ret: Function[] = [];
+    var protoInitializers: Function[];
+    var staticInitializers: Function[];
+    var staticBrand = function (_: any) {
+      return checkInRHS(_) === Class;
+    };
+
+    var existingNonFields = new Map();
+
+    function pushInitializers(initializers: Function[]) {
+      if (initializers) {
+        ret.push(runInitializers.bind(null, initializers));
+      }
+    }
+
+    for (var i = 0; i < decInfos.length; i++) {
+      var decInfo = decInfos[i];
+
+      // skip computed property names
+      if (!Array.isArray(decInfo)) continue;
+
+      var kind = decInfo[1];
+      var name = decInfo[2];
+      var isPrivate = decInfo.length > 3;
+
+      var decoratorsHaveThis = kind & PROP_KIND.DECORATORS_HAVE_THIS;
+      var isStatic = !!(kind & PROP_KIND.STATIC);
+
+      kind &= 7; /* 0b111 */
+
+      var isField = kind === PROP_KIND.FIELD;
+
+      var key = name + "/" + isStatic;
+
+      if (!isField && !isPrivate) {
+        var existingKind = existingNonFields.get(key);
+
+        if (
+          existingKind === true ||
+          (existingKind === PROP_KIND.GETTER && kind !== PROP_KIND.SETTER) ||
+          (existingKind === PROP_KIND.SETTER && kind !== PROP_KIND.GETTER)
+        ) {
+          throw new Error(
+            "Attempted to decorate a public method/accessor that has the same name as a previously decorated public method/accessor. This is not currently supported by the decorators plugin. Property name was: " +
+              name,
+          );
```

Comment:
Q: Do you know how this should work?

Reply:
I don't know, but this message is a bit too long for me, and I plan to modify or remove it later.